packages/main/src/plugin/proxy.ts

```diff
@@ -0,0 +1,132 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ProxySettings, Event } from '@tmpwip/extension-api';
+import type { ConfigurationRegistry, IConfigurationNode } from './configuration-registry';
+import { Emitter } from './events/emitter';
+
+/**
+ * Handle proxy settings for Podman Desktop
+ */
+export class Proxy {
+  private proxySettings: ProxySettings | undefined;
+  private proxyState = false;
+
+  private readonly _onDidUpdateProxy = new Emitter<ProxySettings>();
+  public readonly onDidUpdateProxy: Event<ProxySettings> = this._onDidUpdateProxy.event;
+
+  private readonly _onDidStateChange = new Emitter<boolean>();
+  public readonly onDidStateChange: Event<boolean> = this._onDidStateChange.event;
+
+  constructor(private configurationRegistry: ConfigurationRegistry) {}
+
+  async init(): Promise<void> {
+    const proxyConfigurationNode: IConfigurationNode = {
+      id: 'proxy',
+      title: 'Proxy',
+      type: 'object',
+      properties: {
+        ['proxy.http']: {
+          description: 'Http Proxy value',
+          type: 'string',
+          default: '',
+          hidden: true,
+        },
+        ['proxy.https']: {
+          description: 'Https Proxy value',
```

Comment:
```suggestion
          description: 'Proxy (HTTPS)',
```

Capitalization is a bit odd?

Reply:
I think I did wrongly the rebased will look more closely

Reply:
ah ok no I need to copy from the new wording ;-)

Reply:
fixed